File: javascript/etl/etl.js
var ETL = function() {

}

ETL.prototype.transform = function(old) {
 
  var xformed = {}
 
  // iterate through the properties of the 'old' object
  for (score in old) {
    // iterate through the letters in the array
    for (var i = 0; i < old[score].length; i++) {
      xformed[old[score][i].toLowerCase()] = parseInt(score)
    }
  }
  
  return xformed
}

module.exports = ETL;

File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {

    public Map<String, Integer> transform(Map<Integer, List<String>> old) {
        Map<String, Integer> transformedResult = new HashMap<>();
            old.forEach((key, value) ->
                    value.forEach((item) ->
                            transformedResult.put(item.toLowerCase(), key)));
        return transformedResult;
    }
}
